framework/src/modules/base_module.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable class-methods-use-this */
+
+import { Block } from '@liskhq/lisk-chain';
+import { GenesisBlock } from '@liskhq/lisk-genesis';
+import { GenesisConfig, Consensus, AccountSchema } from '../types';
+import { BaseAsset, StateStore, ReducerHandler } from './base_asset';
+
+interface Reducer {
```

Comment:
Should this be named to match the class property maybe? Like `Reducers` as it would seem it works as a dictionary of reducers. Also should this be Array instead of object to ensure order?